iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/PackageInstallationCacheCheck.java

```diff
@@ -0,0 +1,191 @@
+  private static final Predicate<String> flagWithoutContainingNecessaryFlags = isFlag.and(not(flagWithNecessaryRmOptions));
+
+  private static final CommandDetector APK_ADD = CommandDetector.builder()
+    .with("apk")
+    .withAnyFlagExcept("--no-cache")
+    .with("add")
+    .withAnyFlagExcept("--no-cache")
+    .build();
+
+  private static final CommandDetector APK_CLEAN = buildCleanCacheDetector(Set.of("apk"));
+  private static final CommandDetector REMOVE_APK_CACHE_DETECTOR = buildRemoveCacheDetector(APK_CACHE_LOCATIONS);
+  private static final CommandDetector APT_INSTALL = CommandDetector.builder()
+    .with(APT_COMMANDS)
+    .withAnyFlag()
+    .with("install")
+    .withAnyFlag()
+    .build();
+
+  private static final CommandDetector APT_CLEAN = buildCleanCacheDetector(APT_COMMANDS);
+  private static final CommandDetector REMOVE_APT_CACHE_DETECTOR = buildRemoveCacheDetector(APT_CACHE_LOCATION);
+
+  private static CommandDetector buildCleanCacheDetector(Set<String> commandNames) {
+    var builder = CommandDetector.builder()
+      .with(commandNames);
+
+    if (commandNames.contains("apk")) {
+      builder.with("cache");
+    }
+
+    return builder
+      .with("clean")
+      .build();
+  }
+
+  private static CommandDetector buildRemoveCacheDetector(Set<String> cacheLocations) {
+    return CommandDetector.builder()
+      .with("rm")
+      .withOptionalRepeating(flagWithoutContainingNecessaryFlags)
+      .with(flagWithNecessaryRmOptions)
+      .withAnyFlag()
+      .with(cacheLocations)
+      .build();
+  }
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(RunInstruction.class, PackageInstallationCacheCheck::checkRunInstruction);
+  }
+
+  private static void checkRunInstruction(CheckContext ctx, RunInstruction runInstruction) {
+    List<ArgumentResolution> resolvedArgument = CheckUtils.resolveInstructionArguments(runInstruction);
+
+    SeparatedList<List<ArgumentResolution>, String> splitCommands = ArgumentResolutionSplitter.splitCommands(resolvedArgument);
+
+    List<CommandDetector.Command> sensitiveApkInstallCommands = new ArrayList<>();
+    List<CommandDetector.Command> sensitiveAptInstallCommands = new ArrayList<>();
+
+    for (List<ArgumentResolution> commands : splitCommands.elements()) {
+      analyzeCommands(commands, sensitiveApkInstallCommands, APK_ADD, REMOVE_APK_CACHE_DETECTOR, APK_CLEAN);
+      analyzeCommands(commands, sensitiveAptInstallCommands, APT_INSTALL, REMOVE_APT_CACHE_DETECTOR, APT_CLEAN);
+    }
+
+    sensitiveApkInstallCommands.forEach(command -> ctx.reportIssue(command, MESSAGE));
+    sensitiveAptInstallCommands.forEach(command -> ctx.reportIssue(command, MESSAGE));
+  }
+
+  private static void analyzeCommands(
+    List<ArgumentResolution> commandsToSearchIn, List<CommandDetector.Command> sensitiveInstallCommands,
+    CommandDetector installDetector, CommandDetector removeCacheCommandDetector, CommandDetector cleanCacheCommandDetector) {
```

Comment:
```suggestion
    List<ArgumentResolution> commandsToSearchIn,
    List<CommandDetector.Command> sensitiveInstallCommands,
    CommandDetector installDetector,
    CommandDetector removeCacheCommandDetector,
    CommandDetector cleanCacheCommandDetector) {
```

This might be my personal preference, but for me long parameter lists are more readable when every parameter is on its own line

Reply:
Fine for me either way.